src/classes/BGE_DataImportBatchEntry_CTRL.cls

```diff
@@ -404,24 +436,42 @@ public with sharing class BGE_DataImportBatchEntry_CTRL {
     * to check for potential errors.
     * @return List<DataImport__c> because BDI expects a list
     */
-    private static List<DataImport__c> prepareDataImportForDryRun(DataImport__c dataImportRecord) {
+    private static List<DataImport__c> prepareDataImportsForDryRun(List<DataImport__c> dataImportRecords) {
         // Skip records where user has explicitly selected a matching record in BGE UI
-        if (dataImportRecord.DonationImportStatus__c == Label.bdiMatchedByUser
-                || dataImportRecord.DonationImportStatus__c == Label.bdiMatchedByUserNewOpp
-                || dataImportRecord.DonationImportStatus__c == Label.bdiMatchedApplyNewPayment) {
-            // User defined match, so we do nothing and retain the values
-        } else {
-            // Clear Dry Run fields before running Dry Run so new field values are accounted for
-            dataImportRecord.Status__c = null;
-            dataImportRecord.FailureInformation__c = null;
-            dataImportRecord.DonationImported__c = null;
-            dataImportRecord.DonationImportStatus__c = null;
-            dataImportRecord.PaymentImported__c = null;
-            dataImportRecord.PaymentImportStatus__c = null;
+        for(DataImport__c dataImportRecord : dataImportRecords) {
+            if (dataImportRecord.DonationImportStatus__c == Label.bdiMatchedByUser
+                    || dataImportRecord.DonationImportStatus__c == Label.bdiMatchedByUserNewOpp
+                    || dataImportRecord.DonationImportStatus__c == Label.bdiMatchedApplyNewPayment) {
+                // User defined match, so we do nothing and retain the values
+            } else {
+                // Clear Dry Run fields before running Dry Run so new field values are accounted for
+                dataImportRecord.Status__c = null;
+                dataImportRecord.FailureInformation__c = null;
+                dataImportRecord.DonationImported__c = null;
+                dataImportRecord.DonationImportStatus__c = null;
+                dataImportRecord.PaymentImported__c = null;
+                dataImportRecord.PaymentImportStatus__c = null;
+            }
         }
-        update dataImportRecord;
+        update dataImportRecords;
 
-        return new List<DataImport__c>{dataImportRecord};
+        return dataImportRecords;
+    }
+
+    /*******************************************************************************************************
+    * @description returns a list of all DataImport__c records from a batch Id
+    * @param batchId a Data Import Batch Id
+    * @return List<DataImport__c>
+    */
+    private static List<DataImport__c> getAllDataImportRecordsByBatchId(Id batchId) {
```

Comment:
Could this logic be more shared with the method below? It feels overly repetitive.